brouter-core/src/main/java/btools/router/OsmPath.java

```diff
@@ -346,7 +346,8 @@ protected void addAddionalPenalty(OsmTrack refTrack, boolean detailMode, OsmPath
         if ( rc.startDirectionValid )
         {
           double dir = rc.startDirection.intValue() / CheapRulerSingleton.DEG_TO_RAD;
-          lon0 = lon1 - (int) ( 1000. * Math.sin( dir ) / rc.getCosLat() );
+          double coslat = Math.cos(lat1);

```

Comment:
There was a bug in the last commit as `rc.coslat` is never updated anymore. I've just discarded it as it is not really needed except here. But this should probably be `kx` actually, I'll fix it.